include/mbedtls/check_config.h

```diff
@@ -245,6 +245,9 @@
 #if defined(MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN) &&  !defined(MBEDTLS_HAS_MEMSAN)
 #error "MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN requires building with MemorySanitizer"
 #endif
+#if defined(MBEDTLS_HAS_MEMSAN) && defined(MBEDTLS_HAVE_ASM)
+#error "MEMSAN does not support assembly implementation"
```

Comment:
“Memsan” is our own abbreviation. The rest of the world knows it as [MemorySanitizer or MSan](https://github.com/google/sanitizers/wiki/MemorySanitizer). Please use “MSan” or “MemorySanitizer” in the error message.